File: 930.java
class Solution {
    public int numSubarraysWithSum(int[] A, int S) {
        int res = 0;
        int[] map = new int[A.length+1];
        map[0] = 1;
        int sum = 0;
        for(int e:A){
            sum+=e;
            if(sum>=S)
                res+=map[sum-S];
            map[sum]++;
        }
        return res;
    }
}

File: 930.py
class Solution:
    def numSubarraysWithSum(self, A: List[int], S: int) -> int:
        odd = [0] * (len(A)+1)
        odd[0] = 1
        cnt, ans = 0, 0
        for num in A:
            cnt += num
            if cnt >= S:
                ans += odd[cnt-S]
            odd[cnt] += 1
        print(odd)
        return ans